lib/dolly/document.rb

```diff
@@ -1,9 +1,13 @@
 require "dolly/query"
 require "dolly/property"
+require 'dolly/timestamps'
+require 'active_model/callbacks'
 
 module Dolly
   class Document
     extend Dolly::Connection
+    extend ActiveModel::Callbacks
```

Comment:
Sup guys - 

Isn't this the same path we went down with `couch_rest_adapter` ? ActiveModel will suck you in, which is fine, but I guess it really depends on your desired purpose.

ex) https://github.com/amco/couch_rest_adapter/blob/f366e14c3864c346e6d01debb435a64966cae805/lib/couch_rest_adapter.rb#L17-L20